Git-cheatsheet.md

```diff
@@ -0,0 +1,71 @@
+# Git CheatSheet
+
+We use git to version control (save) our changes. This allows everyone to receive the latest changes that have been agreed by everyone. Here is a simple cheatsheet to follow to save your changes.
+
+### Pushing your changes
+
+```
+git checkout -b <branch_name>
+git status
+git add <file>
+git commit -m "<message>"
+git push origin <branch>
+```
+
+Now head over to github.com to raise a pull request https://github.com/ministryofjustice/robot-framework-test-ccms. You'll see a yellow message asking you if you'd like to.
+
+Click on "Compare & Pull Request" against your branch. On the next screen, add details of what your changes were and then click on "Create pull request button". This is now ready to be revieved by someone who can comment and approve the pull request.
+
+Action: Someone else needs to review your request. Fling it in chat or ask in stand ups if someone is there to review.
+
+They would review, and either approve or ask for changes. This is done by going into the files tab, and clicking on review button. To leave comments and feedback, you can click on the line and leave comments.
+
+### Once approved
+
+Once approved, Go to the pull request on Github.com and click on Merge. This will merge your changes into main and everyone will now be using it as part of the main software distribution. At this stage, it is safe to delete your branch (Standard practice is to delete your branch once merged in, it can be recovered later if you need it).
+
+### To pull changes
+
+```
+git checkout main
+git pull origin main
+```
+
+An Example
+===
+
+I have made changes to a file `robot_scripts/Tasks/login.robot` and `robot_scripts/Support/ebs_helper.robot` file. I'd like to only save changes of the `robot_scripts/Support/ebs_helper.robot` file and NOT `robot_scripts/Tasks/login.robot`.
+
+### On my machine
+```
+# On my machine
+git checkout -b update-ebs_helper
+git status
+git add robot_scripts/Support/ebs_helper.robot
+git commit -m "Fixed EBS helper method that identifies window titles"
+git push origin update-ebs_helper
+```
+
+### On Github.com
+1. Raise a pull request
+2. Ask someone within the team to review please (Slack/Stand ups).
+3. Once approved, merge it in on Github.com and delete the "update-ebs_helper" branch.
+
+### On my machine
+```
+git checkout main
+git pull origin main
+```
+
+Further details
+===
+
+| Command | Description |
+|---------|-----------|
+| git checkout | The `git checkout` command above will create a new branch for you to save your work on. This is a little like creating a duplicate file from an existing file to preserve the original file. You usually only need a new branch for the scope of a particular ticket. We don't recommend reusing the same branch as it promotes scoping your changes for the ticket at hand and keeping your changes manageable for reviewing later on. |
```

Comment:
Could also mention that `git checkout` can be used for other things such as switching between existing branches. No need to go into detail but just to point out it has more than one use.

Reply:
Nice one, done.